website/docs/docs/contributing/testing-a-new-adapter.md

```diff
@@ -0,0 +1,449 @@
+- The file name must begin with `test_`
+- The class must begin with `Test`
+- All test functions must begin with `test_`
+
+<File name="tests/functional/example/test_example.py">
+
+```python
+import pytest
+from dbt.tests.util import run_dbt
+
+# our file contents
+from tests.functional.example.fixtures import (
+    my_seed_csv,
+    my_model_sql,
+    my_model_yml,
+)
+
+# class must begin with 'Test'
+class TestExample:
+    """Methods below are of two types:
+        - fixtures defining the setup for this test case
+          (scoped to the class, and reused for all tests in the class)
+        - actual tests, whose names begin with 'test_'
+    """
+    
+    # configuration in dbt_project.yml
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {"name": "example"}
+
+    # everything that goes in the "seeds" directory
+    @pytest.fixture(scope="class")
+    def seeds(self):
+        return {
+            "my_seed.csv": my_seed_csv,
+        }
+
+    # everything that goes in the "models" directory
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "my_model.sql": my_model_sql,
+            "my_model.yml": my_model_yml,
+        }
+    
+    # the method defines the actual sequence of steps to execute
+    # pytest will take care of all "setup" + "teardown"
+    def test_run_seed_test(self, project):
+        """Seed, then run, then test. We expect one of the tests to fail
+           An alternative pattern is to use pytest "xfail" (see below)
+        """
+        # seed seeds
+        results = run_dbt(["seed"])
+        assert len(results) == 1
+        # run models
+        results = run_dbt(["run"])
+        assert len(results) == 1
+        # test tests
+        results = run_dbt(["test"], expect_pass = False) # expect failing test
+        assert len(results) == 2
+
+    @pytest.mark.xfail
+    def test_build(self, project):
+        """Expect a failing test"""
+        # do it all
+        results = run_dbt(["build"])
+```
+
+Then, we invoke `pytest`:
+
+```sh
+$ python3 -m pytest tests/functional/test_example.py
+=========================== test session starts ============================
+platform ... -- Python ..., pytest-..., pluggy-...
+rootdir: ...
+plugins: ...
+
+tests/functional/test_example.py .X                                  [100%]
+
+======================= 1 passed, 1 xpassed in 1.38s =======================
+```
+
+</File>
+
+The [pytest usage docs](https://docs.pytest.org/how-to/usage.html) offer guidance and a full command reference. In our experience, it can be particularly helpful to use the `-s` flag (or `--capture=no`) to print logs from the underlying dbt invocations, and to step into an interactive debugger if you've added one. You can also use environment variables to set [global dbt configs](global-configs), such as `DBT_DEBUG` (to show debug-level logs).
+
+## Testing your adapter
```

Comment:
I'm flexible on this title just trying to capture what this section involves and it's more background info than steps for testing your adapter.
```suggestion
## Understanding adapter testing
```